lib/assert.js

```diff
@@ -309,10 +323,26 @@ assert.doesNotThrow = function doesNotThrow(block, error, message) {
       expected: error,
       operator: 'doesNotThrow',
       message: `Got unwanted exception${details}\n${actual && actual.message}`,
-      stackStartFn: doesNotThrow
+      stackStartFn
     });
   }
   throw actual;
+}
+
+assert.throws = function throws(block, error, message) {
+  expectsError(getActual(block), error, message, this);
```

Comment:
I am a bit surprised that `this` for the `stackStartFn` shall work. `this` should normally refer to either `assert.ok` or `assert.strict`.
Can you please write a test for this as well if none exist? :-)

Reply:
I added a test, and you were absolutely right.
Providing explicit values keeps the existing stack-traces:

https://github.com/feugy/node/blob/6cdbbe6f520862edda962dbf8507c9e7917bdd1e/test/parallel/test-assert-async.js#L60-L98